### ISSUE TITLE
Not good for detecting small objects ?

### ISSUE THREAD

**Issue by trungpham2606** [ruinmessi/ASFF#21]
@ruinmessi 
Thanks for sharing such a great work.
Iam trying your work with a dataset which every objects in the image are quite small.
Iam using 1 GPU for training only and the model trained without any errors. But the loss didnt decrease.
Maybe the problem comes from the fact that I didnt use so many methods that you had used for training this network such as synchronized batch normalization, FP16 training ...
But Iam wondering whether it is the main problem or this network is struggle with detecting small objects ?

Here is 1 image in the dataset that Iam using for training.
![1](https://user-images.githubusercontent.com/35657426/69837251-5d616800-1280-11ea-940e-cbf19d53232f.PNG)

**Comment by trungpham2606** [comment]
@ruinmessi 1 more thing weird is that when I used the weights that I had trained, if I set the threshold score in range (0, 1) as usual, no predictions were shown. But when I set it 10000 smaller, the results appeared.
![image](https://user-images.githubusercontent.com/35657426/69842502-17af9a00-1296-11ea-9889-e0cdf22ca3af.png)

**Comment by ruinmessi** [comment]
Sorry, I have no idea about your dataset and training setting. As the img you showed is quite simple, I suggest you use a ''shallow'' model, like SSD,  RFBNet for this task. They are more friendly for single gpu training, and the performance may be not too bad.

**Comment by trungpham2606** [comment]
@ruinmessi I know that my dataset is quite simple but the results I got from your model are not good. I had used Centernet before and I want to have a comparison between 2 models in some simple dataset first.

**Comment by trungpham2606** [comment]
@ruinmessi And tried to train the network again with coco 2014 valid, and it still detects very well. So I think this network is not good at detecting small objects as the fact that yolo itself is not good at handling that problem.
I also saw that in yolo_head you defined 9 anchors. 
Do those anchors size effect the network when is trained for detecting small objects ?

**Comment by wyj18502986957** [comment]
@trungpham2606  I need your test.py code for detecting the single picture,can you send it to me at my email 18502986957@163.com ,thank you

**Comment by trungpham2606** [comment]
@wyj18502986957 yeah, sure bro.

**Comment by uiui1999vn** [comment]
@trungpham2606 I also need that test.py code, can you send it to me at my email uiui1999vn@gmail.com

**Comment by iodncookie** [comment]
@trungpham2606 , @wyj18502986957 I want detecting my dataset,can you send the test.py code to me .my email is tionponchine@126.com. thanks

**Comment by wushilian** [comment]
@trungpham2606 What's the effect of centernet？

**Comment by trungpham2606** [comment]
@wushilian what do you mean "effect" ?
